mlflow/spark.py

```diff
@@ -198,11 +207,11 @@ def log_model(
             input_example=input_example,
             await_registration_for=await_registration_for,
         )
-    # If Spark cannot write directly to the artifact repo, defer to Model.log() to persist the
-    # model
     model_dir = os.path.join(run_root_artifact_uri, artifact_path)
+    # Try to write directly to the artifact repo via Spark. If this fails, defer to Model.log()
+    # to persist the model
     try:
-        spark_model.save(os.path.join(model_dir, _SPARK_MODEL_PATH_SUB))
+        spark_model.save(posixpath.join(model_dir, _SPARK_MODEL_PATH_SUB))
```

Comment:
Note: should be a posixpath join since HDFS URIs are posixpath style (e.g. we don't want to include backward slashes `\` here on platforms like Windows)